### DISCUSSION TITLE
Handle bad submissions in a plugin

### DISCUSSION THREAD

**Discussion by oxedions** [nextflow-io/nextflow#3908] [Q&A] [answered]
Hello 🙂 

I am developing a plugin (https://github.com/nimbix/nf-jarvice/) for my company platform.
I was wondering if the current behavior for tasks submission is the correct one.

In the current plugin state, when a task is submitted, if submission failed, I have a log.error message, and submit() void do not end with             tasks status as SUBMITTED.

When this happens, nextflow then just hangs, waiting for user to kill the current process (Ctrl+c). Is it the expected behavior ? I find logical not to stop nextflow, because there could be other tasks running, and it needs to gracefully end all of them. But in that case, what would be the way to inform nextflow that all tasks should now be terminated and current pipeline stopped since one of the tasks failed to submit ? (assuming there is no "retry" mechanism).

Many thanks in advance 😊 

Relevant part of the code is here: https://github.com/nimbix/nf-jarvice/blob/b21120f9385a5aec59b317262ccee613236147af/plugins/nf-jarvice/src/main/nextflow/jarvice/JarviceTaskHandler.groovy#L103

Ox

**Comment by pditommaso** [comment]
Instead of logging the error I think you have two choices: 
1. Throw an exception that likely will cause the execution to stop 
2. Mark the task FAILED that should allow nextflow to handle the task an any other failure

**Reply by oxedions** [reply]
Dear @pditommaso 

Many thanks for this answer !

Solution 2 would be I think the best, so nextflow can garbage collect possible other running tasks.

However, I fail to find how to set a task as failed.
When investigating nextflow.processor.TaskStatus, I only see :

- COMPLETED 
- NEW
- RUNNING
- SUBMITTED

How can I mark, in submit void, a task as failed to be submitted ?

Should I set it as COMPLETED with an error exit code instead ?

```
            this.status = TaskStatus.COMPLETED
            task.exitStatus = 1
```

**Reply by pditommaso** [reply, marked as answer]
Should be enough the `this.status = TaskStatus.COMPLETED`. The absence of exitStatus is considered an error condition

**Reply by oxedions** [reply]
Dear @pditommaso 

I will set it this way then, so it prevents any more submissions 🙂

Thank you very much for this answer!
